### ISSUE TITLE
Feature request: Upload external album art

### ISSUE THREAD

**Issue by benni3005** [thebigmunch/gmusicapi-scripts#34]
Hi there,

is it planned to upload external album art in the future? I like the sync feature a lot, but it's a pain to manually upload the album art to Google Play Music. Especially for libraries containing hundreds of albums.

I know, there's nothing in the unofficial API yet (see issue simon-weber/Unofficial-Google-Music-API#242), but if it were, could you include this feature?

Greetings,
Benni

**Comment by thebigmunch** [comment]
Yes, this is something that I was always going to add as, at least, an option when the gmusicapi Musicmanager interface got support for it rather than implementing it in my wrapper. But if I do get enough time soon to work on things, I may just do it myself either in the wrapper or in gmusicapi itself.

**Comment by porkhelmet** [comment]
Hey all, I actually had this feature working with gmusicapi-scripts and the gmusicapi Webclient for a whole 7 days before app specific passwords stopped working.  I posted some of the code I used in simon-weber/gmusicapi#352.  I am not skilled enough to fix the authentication issues so hopefully the art extraction and folder art search code will aid your efforts.  Thanks for creating these awesome scripts!

**Comment by thebigmunch** [comment]
The basic structure of grabbing artwork is rather trivial. Unfortunately, there are many common cases that your code doesn't cover that need to be.

**Comment by porkhelmet** [comment]
As always, you are on the money Munch.  The code was only designed and tested for my music collection, so the scope was pretty limited.  So many features, so little time!

**Comment by thebigmunch** [comment]
Just to mention, this may be the start of a config file. There are likely too many variations of preferences to try to be smart about it.

**Comment by thebigmunch** [comment]
Alright, time to gather some info and get goin' on this. Since the actual meat of this would be done in my wrapper, see the [appropriate issue](https://github.com/thebigmunch/gmusicapi-wrapper/issues/6) on that repo for more discussion/feedback. This issue will stay open until I add the CLI option to the scripts.

**Comment by thebigmunch** [comment]
Good news everybody, it's happening!!!

I played around with the Music Manager tonight, and it turned out to be trivial on the protocol end. I have to flesh it out properly, but I should be adding album art uploading to gmusicapi in the next week or two. Then I'll be able to implement album art uploading on this end. I'll try to think about any smart automatic ways to make it simpler, but there will probably be an option added to specify an image and/or a list of patterns in priority order.

**Comment by sprestwood** [comment]
Yay, that's really great to hear! Thanks for following up with this! :)

**Comment by thebigmunch** [comment]
So, there was a snag. You can read the details [here](https://github.com/simon-weber/gmusicapi/issues/242#issuecomment-248083213). Hopefully I can get back to solving this soon : (
